internal/runners/initialize/init.go

```diff
@@ -50,184 +47,138 @@ func New(prime primeable) *Initialize {
 // (i.e. `state use show`).
 // Error handling is not necessary because it's an input error to not include a language to
 // `state init`. We're just trying to infer one as a convenience to the user.
-func inferLanguage(config projectfile.ConfigGetter) string {
+func inferLanguage(config projectfile.ConfigGetter) (string, string, bool) {
 	defaultProjectDir := config.GetString(constants.GlobalDefaultPrefname)
 	if defaultProjectDir == "" {
-		return ""
+		return "", "", false
 	}
 	defaultProj, err := project.FromPath(defaultProjectDir)
 	if err != nil {
-		return ""
+		return "", "", false
 	}
 	commitID := defaultProj.CommitUUID()
 	if commitID == "" {
-		return ""
+		return "", "", false
 	}
 	lang, err := model.FetchLanguageForCommit(commitID)
 	if err != nil {
-		return ""
+		return "", "", false
 	}
-	return fmt.Sprintf("%s@%s", lang.Name, lang.Version)
+	return lang.Name, lang.Version, true
 }
 
-func sanitize(params *RunParams, config projectfile.ConfigGetter) error {
-	if params.Language == "" {
-		params.Language = inferLanguage(config)
+func (r *Initialize) Run(params *RunParams) error {
+	if err := params.Namespace.Validate(); err != nil {
```

Comment:
I could be remembering wrong; but I think captain already does this `Validate()` call, so this would be redundant.